### PR TITLE
fix: migration compatibility

### PR DESCRIPTION
**Pull request by andrej-jasso**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,112 @@ DerivedData/
 .swiftpm/config/registries.json
 .swiftpm/xcode/package.xcworkspace/contents.xcworkspacedata
 .netrc
+
```
Comment (andrej-jasso) [inline .gitignore:10]: I was getting xcuser items into diffs. Added our standard gitignore.

File: GoodPersistence-Sample/GoodPersistenceKeychainTests/MajorMigrationTestsV1/KeychainItemAccessibility.swift
Comment (andrej-jasso) [inline GoodPersistence-Sample/GoodPersistenceKeychainTests/MajorMigrationTestsV1/KeychainItemAccessibility.swift:1]: Keychain from version 1 of the library to compare and make migration tests

File: Tests/GoodPersistenceTests/MigrationTests/UserDefaultsWrapperV1.swift
Comment (andrej-jasso) [inline Tests/GoodPersistenceTests/MigrationTests/UserDefaultsWrapperV1.swift:1]: User defaults from version one to compare and make tests

File: Sources/GoodPersistence/KeychainValue.swift [hunk 2]
```diff
@@ -27,7 +28,21 @@ public struct KeychainConfiguration {
     let protocolType: ProtocolType?
     let accessGroup: String?
     let authenticationType: AuthenticationType?
-    
+
+    init(
```
Comment (andrej-jasso) [inline Sources/GoodPersistence/KeychainValue.swift:32]: Custom init for default nil values

File: Sources/GoodPersistence/LoggingMonitor/LoggingPersistenceMonitor.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
+//
+//  LoggingPersistenceMonitor.swift
+//
+//  Created by Andrej Jasso on 12/04/2024.
+//
+
+import Foundation
+
+public final class LoggingPersistenceMonitor: PersistenceMonitor {
```
Comment (andrej-jasso) [inline Sources/GoodPersistence/LoggingMonitor/LoggingPersistenceMonitor.swift:9]: Prototype logging monitor

Comment (plajdo) [reply Sources/GoodPersistence/LoggingMonitor/LoggingPersistenceMonitor.swift:9]: @andrej-jasso tieto logging monitory by nesli nejako zlucit s loggermi v GoodNetworkingu? (mam na mysli OSLogLogger/PrintLogger)

Comment (andrej-jasso) [reply Sources/GoodPersistence/LoggingMonitor/LoggingPersistenceMonitor.swift:9]: GoodLogger? :D

**Pull request merged**